assets/src/settings-page/index.js

```diff
@@ -161,7 +222,7 @@ domReady( () => {
 		render( (
 			<ErrorContextProvider>
 				<Providers>
-					<Root optionsRestPath={ OPTIONS_REST_PATH } />
+					<Root section={ global.location.hash.replace( /^#/, '' ) } />
```

Comment:
I think this can be improved… namely, if I am already on the AMP Settings screen and click the admin menu item, then I am jumped down the screen without the section expanding. Therefore, it would be preferable if there was also a [`hashchange`](https://developer.mozilla.org/en-US/docs/Web/API/Window/hashchange_event) event that would re-render this tree whenever it fires. This will ensure that the section always reflects what is currently in the `location.hash`.

Reply:
I did this in 754aadd8572de9744af0f99953b077c3bded9bf9 but it was surprisingly complicated to get it to work in all the scenarios (initial load of the page, changing the hash in the URL location bar, clicking the Analytics link in the menu), and I'm not sure how I feel about how much code I had to add. It's working, though.